extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/quantiles/QuantilesSketchAggregatorFactory.java

```diff
@@ -0,0 +1,327 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.datasketches.quantiles;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.primitives.Doubles;
+import com.google.common.primitives.Ints;
+import com.metamx.common.StringUtils;
+import com.yahoo.sketches.Util;
+import com.yahoo.sketches.quantiles.QuantilesSketch;
+import com.yahoo.sketches.quantiles.QuantilesSketchBuilder;
+import com.yahoo.sketches.quantiles.Union;
+import io.druid.query.aggregation.Aggregator;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.query.aggregation.AggregatorFactoryNotMergeableException;
+import io.druid.query.aggregation.BufferAggregator;
+import io.druid.segment.ColumnSelectorFactory;
+import io.druid.segment.ObjectColumnSelector;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class QuantilesSketchAggregatorFactory extends AggregatorFactory
+{
+  private static final byte CACHE_TYPE_ID = 16;
+
+  public static final int DEFAULT_MAX_SKETCH_SIZE = 1024;
+
+  protected final String name;
+  protected final String fieldName;
+  protected final int size;
+  private final boolean shouldFinalize;
+  private final boolean isInputSketch;
+
+  public static final Comparator<QuantilesSketch> COMPARATOR = new Comparator<QuantilesSketch>()
+  {
+    @Override
+    public int compare(QuantilesSketch o, QuantilesSketch o1)
+    {
+      return Doubles.compare(o.getN(), o1.getN());
+    }
+  };
+
+  @JsonCreator
+  public QuantilesSketchAggregatorFactory(
+      @JsonProperty("name") String name,
+      @JsonProperty("fieldName") String fieldName,
+      @JsonProperty("size") Integer size,
+      @JsonProperty("shouldFinalize") Boolean shouldFinalize,
```

Comment:
What is the purpose of "shouldFinalize"?  If you don't want things finalized you can already just send your query with `"finalize": false` in the context.